docs/proposals/cloud-provider-refactoring.md

```diff
@@ -0,0 +1,106 @@
+## Refactor Cloud Provider out of Kubernetes Core
+
+As kubernetes has evolved tremendously, it has become difficult for different cloudproviders (currently 7) to make changes and iterate quickly. Moreover, the cloudproviders are constrained by the kubernetes build/release lifecycle. This proposal aims to move towards a kubernetes code base where cloud providers specific code will move out of the core repository, and will be maintained by the cloud providers themselves.
+
+### 1. Current use of Cloud Provider
+
+The following components have cloudprovider dependencies
+
+    1. kube-controller-manager
+    2. kubelet
+    3. kube-apiserver
+
+##### Cloud Provider in Kube-Controller-Manager
```

Comment:
These headings get really small. What about bumping up each tier so that this one is ### instead of ####?